RecoTracker/TkSeedingLayers/src/HitExtractorSTRP.cc

```diff
@@ -271,11 +314,33 @@ HitExtractor::Hits HitExtractorSTRP::hits(const TkTransientTrackingRecHitBuilder
       if (skipClusters)
         cleanedOfClusters(ttrhBuilder, ev, result, false, cleanFrom);
     }
+    if (hasVectorHits) {
+      LogTrace("HitExtractorSTRP") << "Getting vector hits for IdLayer " << theIdLayer;
+      edm::Handle<VectorHitCollectionNew> vectorHits;
+      ev.getByToken(theVectorHits, vectorHits);
+      //FIXME: check the skipClusters with VHits
+      if (skipClusters)
+        cleanFrom = result.size();
+      auto getter = tTopo->tidDetIdWheelComparator(static_cast<unsigned int>(theSide), theIdLayer);
+      VectorHitCollection::Range range = vectorHits->equal_range(getter.first, getter.second);
+      for (VectorHitCollection::const_iterator it = range.first; it != range.second; ++it) {
+        int ring = tTopo->tidRing(it->detId());
+        if (!ringRange(ring))
+          continue;
+        for (VectorHitCollection::DetSet::const_iterator hit = it->begin(), end = it->end(); hit != end; ++hit) {
```

Comment:
just use `vector::insert()` (no need for loop)

Reply:
I don't really understand the suggestion here.

Reply:
`result` is a vector. rather than emplacing each entry from `it` one at a time, you can just do: `result.insert(result.end(),it->begin(),it->end());`

Reply:
Mhhh, trying this I get this error:

`no known conversion for argument 1 from 'VectorHit' to 'mayown_ptr<BaseTrackerRecHit>&&'`

Reply:
Ah, it's upcasting in the `push_back()` statement. Then I think it has to stay the way it is.